compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/nested-optional-chains.expect.md

```diff
@@ -0,0 +1,229 @@
+
+## Input
+
+```javascript
+import {identity} from 'shared-runtime';
+
+/**
+ * identity(...)?.toString() is the outer optional, and prop?.value is the inner
+ * one.
+ * Note that prop?.
+ */
+function useFoo({
+  prop1,
+  prop2,
+  prop3,
+  prop4,
+  prop5,
+  prop6,
+}: {
+  prop1: null | {value: number};
+  prop2: null | {inner: {value: number}};
+  prop3: null | {fn: (val: any) => NonNullable<object>};
+  prop4: null | {inner: {value: number}};
+  prop5: null | {fn: (val: any) => NonNullable<object>};
+  prop6: null | {inner: {value: number}};
+}) {
+  // prop1?.value should be hoisted as the dependency of x
+  const x = identity(prop1?.value)?.toString();
+
+  // prop2?.inner.value should be hoisted as the dependency of y
```

Comment:
is it expected that we're not actually inferring this as the dep? the actual inferred dep is just `prop2?.inner` (no .value)

Reply:
@josephsavona This fixture had different output when toggling `@enablePropagateDepsInHIR`, so it lives in both `propagate-scope-hir-fork` and the base fixtures directory. This is the non hir-rewrite version

Will make sure to leave a PR comment next time I add fixtures whose output conflict with comments. Sorry for the confusion!

Reply:
ahhhh got it